scripts/convert_tags_to_custom_properties/convert_tags_to_custom_properties.py

```diff
@@ -0,0 +1,206 @@
+import os
+import requests
+import json
+
+OPSLEVEL_API_TOKEN = os.environ["OPSLEVEL_API_TOKEN"]
+OPSLEVEL_ENDPOINT = "https://app.opslevel.com/graphql"
+
+LIST_CUSTOM_PROPERTIES_QUERY = """
+    query custom_service_properties($endCursor:String) {
+      account {
+        propertyDefinitions(after: $endCursor) {
+          pageInfo{
+            hasNextPage
+            endCursor
+          }
+          nodes {
+            name
+            aliases
+            schema
+            id
+          }
+        }
+      }
+    }
+"""
+
+SERVICES_BY_TAG_QUERY = """
+    query services_by_tag($endCursor: String, $tag_key:String){
+      account{
+        services(tag: {key: $tag_key}, after: $endCursor){
+          pageInfo{
+            hasNextPage
+            endCursor
+          }
+          nodes{
+            id
+            tags{
+              nodes{
+                id
+                key
+                value
+              }
+            }
+          }
+        }
+      }
+    }
+"""
+
+UPDATE_PROPERTY_MUTATION = """
+    mutation update_property($service_id:ID, $definition_alias:String, $value:JsonString!){
+      propertyAssign(input: {owner: {id: $service_id}, definition: {alias: $definition_alias},
+      value: $value, runValidation: false}) {
+        property{
+          value
+          owner{
+            ...on Service{
+              name
+            }
+          }
+        }
+        errors{
+          message
+          path
+        }
+      }
+    }
+"""
+
+
+def opslevel_graphql_query(query, variables=None):
+    headers = {
+        "Content-Type": "application/json",
+        "Authorization": f"Bearer {OPSLEVEL_API_TOKEN}",
+    }
+    data = {"query": query, "variables": variables}
+    response = requests.post(OPSLEVEL_ENDPOINT, json=data, headers=headers)
```

Comment:
What happens if this gets a non-200 status code? for example user could get `429` if they are getting timed out or `400` if their auth is bad

Reply:
added `raise Exception` to improve this.